pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -20,25 +19,38 @@
   xorg,
 }:
 
-stdenvNoCC.mkDerivation (finalAttrs: {
+let
   pname = "atlauncher";
   version = "3.4.36.10";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-JZTiYcea5ik8a4RmNLxZcuea7spGWftUGRiRW2Ive7c=";
+  src = fetchFromGitHub {
+    owner = "ATLauncher";
+    repo = "ATLauncher";
+    rev = "v${version}";
+    hash = "sha256-EMV/M5W/acJaQJsi4COoRQC2YVfuJG9HGcwvGfakoNo=";
   };
+in
+stdenvNoCC.mkDerivation {
+  inherit pname version src;
 
-  env.ICON = fetchurl {
-    url = "https://atlauncher.com/assets/images/logo.svg";
-    hash = "sha256-XoqpsgLmkpa2SdjZvPkgg6BUJulIBIeu6mBsJJCixfo=";
+  nativeBuildInputs = [
+    gradle
+    makeWrapper
+  ];
+
+  mitmCache = gradle.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
   };
 
-  dontUnpack = true;
+  doCheck = true;
 
-  nativeBuildInputs = [
-    copyDesktopItems
-    makeWrapper
+  gradleFlags = [
+    "build"
```

Comment:
You didn't apply this

```suggestion
```

Reply:
Ah, I don't need the build flag? Sorry, I misread the previous review as just adding `doCheck`.

Reply:
"build" isnt a flag, it's the task to build, and if you want to set it, you have to set `gradleBuildTask`. By default the setup hook sets `gradleBuildTask` to `assemble`, and the `build` task is usually `assemble` + `test` + potentially some other stuff. That's why I recommended removing this flag and simply setting `doCheck` to `true`.